### DISCUSSION TITLE
Huge thumbnail count keeps increasing - is this normal?

### DISCUSSION THREAD

**Discussion by Pandamonium42** [immich-app/immich#9054] [Q&A] [answered]
Hi All,

New user here. I have recently set Immich up in an Unraid docker and it looks really good. I've tried several photo management tools and this one looks like it meets all my needs so far. I installed version v1.102.2 (or maybe 1.102.1) a couple of weeks ago and started it processing my photos. It has been doing so ever since. I accept that this is going to be a long process due to the number of photos I have so that is not a problem.

I just wanted to check that what I am seeing in my setup is normal please. I have 458,000 photos but the number of Thumbnails waiting for processing keeps increasing despite the continuous processing going on. At the moment, the backlog is now 2,416,000 and still increasing. I see that it is creating large, small and blurred thumbnails so that is 3 per photo, plus faces I assume but this number is now looking a bit high to me.

Is this normal or should I start looking for problems?

Many Thanks

**Comment by mertalev** [comment]
If you have a lot of live photos, the motion part of the image will be queued and skipped when the job is processed. The queue doesn't necessarily translate to that many thumbnails being generated.

**Reply by Pandamonium42** [reply]
Thanks for the information, but I don't have many (if any) live photos. They are mostly Canon raw files (CR2 & CR3).

I didn't know that some queued items can get skipped though so maybe I just need to be more patient.

**Comment by bo0tzz** [comment]
If you're using an external library, double check that you haven't included the UPLOAD_LOCATION in it. If you did there'll be an infinite loop of thumbnail generation for thumbnails etc.

**Reply by Pandamonium42** [reply]
I am using an external library but fortunately my upload is not within that folder. I assumed this was true as I thought my file count would also keep rising and it doesn't, but I went and checked anyway.

While I was there I took a look at my thumbs folder and it already contains nearly 900,000 thumbnails so the full total (existing thumbnails plus queued) looks like it will be well over 3,000,000. I have no issue with that if it's what is needed, I just wanted to check I don't have a problem somewhere.

Everything seems to be going OK so I'll leave it alone and keep checking it in the hope the queue starts dropping eventually.

**Comment by Pandamonium42** [comment, marked as answer]
Over the weekend the thumbnail queue finally started to decrease. Looks like there were no issues, just a higher count than I expected.
